frontend/src/pages/MatchingGame.jsx

```diff
@@ -0,0 +1,118 @@
+            alert("Matching Game requires at least 4 questions.");
+            setMode(null);
+            setSelectedTest(null);
+        } else {
+            setShuffledQuestions([...test.questions].sort(() => Math.random() - 0.5));
+            setShuffledAnswers([...test.questions].sort(() => Math.random() - 0.5));
+        }
+    }, [test, setMode, setSelectedTest]);
+
+    const handleQuestionClick = (question) => {
+        setSelectedQuestion(question);
+        setErrorMessage("");
+    };
+
+    const handleAnswerClick = (answer) => {
+        setSelectedAnswer(answer);
+        setErrorMessage("");
+    };
+
+    const checkMatch = () => {
+        if (selectedQuestion && selectedAnswer) {
+            if (selectedQuestion.answer === selectedAnswer.answer) {
+                setMatchedPairs([...matchedPairs, selectedQuestion.question]);
+                setSelectedQuestion(null);
+                setSelectedAnswer(null);
+            } else {
+                setErrorMessage("Incorrect match! Try again.");
+            }
+        }
+    };
+
+    const isMatched = (item) =>
+        matchedPairs.includes(item.question || item.answer);
+
+    return (
+        <div className="bg-[#0f172a] min-h-screen flex flex-col items-center py-12 px-6 text-white">
+            <h1 className="text-4xl font-bold mb-8">Matching Game</h1>
+
+            <div className="grid grid-cols-4 gap-6 w-full max-w-4xl">
+                {/* Pytania */}
+                {shuffledQuestions.map((question, index) => (
+                    <div
+                        key={index}
+                        className={`p-4 rounded-lg cursor-pointer text-center bg-blue-600 ${
+                            isMatched(question) ? "opacity-50 pointer-events-none" : ""
+                        } ${
+                            selectedQuestion === question
+                                ? "ring-4 ring-yellow-500"
+                                : ""
+                        }`}
+                        onClick={() => handleQuestionClick(question)}
+                    >
+                        {question.question}
+                    </div>
+                ))}
+
+                {/* Odpowiedzi */}
+                {shuffledAnswers.map((answer, index) => (
+                    <div
+                        key={index}
+                        className={`p-4 rounded-lg cursor-pointer text-center bg-green-600 ${
+                            isMatched(answer) ? "opacity-50 pointer-events-none" : ""
+                        } ${
+                            selectedAnswer === answer
+                                ? "ring-4 ring-yellow-500"
+                                : ""
+                        }`}
+                        onClick={() => handleAnswerClick(answer)}
+                    >
+                        {answer.answer}
+                    </div>
+                ))}
+            </div>
+
+            {/* Komunikaty */}
```

Comment:
Let's change the comment to English.
```suggestion
            {/* Prompts */}
```